dlc-manager/src/sub_channel_manager.rs

```diff
@@ -1753,20 +1739,53 @@ where
         &self,
         sub_channel_finalize: &SubChannelFinalize,
         counter_party: &PublicKey,
-    ) -> Result<(), Error> {
-        self.ln_channel_manager.with_useable_channel_lock(
+    ) -> Result<SubChannelRevoke, Error> {
+        let channel_details = self
+            .ln_channel_manager
+            .get_channel_details(&sub_channel_finalize.channel_id)
+            .ok_or_else(|| {
+                Error::InvalidParameters(format!(
+                    "Unknown LN channel {:02x?}",
+                    sub_channel_finalize.channel_id
+                ))
+            })?;
```

Comment:
Can't we get the channel details inside of the channel lock closure below?

Reply:
As is it would deadlock. I thought about adding a way to get the channel details from the callback but it didn't feel that important for now.